### PR TITLE
Refactored into classes and adding stopping rules

### PR DESCRIPTION
**Pull request by Rambatino**
Not sure if splitting the classes out in this PR should really happen, but the file was becoming gigantic and I was adding another method.

### CODE REVIEW

File: tests/setup_tests.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
 sys.path.append(ROOT_FOLDER)
 
 import CHAID
-
+import CHAID.chaid_vector as CHAIDVector
```
Comment (xulaus) [inline tests/setup_tests.py]: This seems wrong. I think the __init__.py should change rather than how you import stuff

**Comment by xulaus** [comment]
Im not sure I get what rules is doing, could you either reword the doc sting or add some explanatory comments?

File: tests/setup_tests.py (hunk 1 above)
Comment (Rambatino) [inline tests/setup_tests.py]: Okya, will sort that out today

**Comment by Rambatino** [comment]
Cool yeah. It was a way I could avoid traversing the tree. 
Essentially, if you capture the heterogeneity in the respondents of that tree, (i.e. get the uniq rows) you then have all the different unique attributes that define that node

**Pull request closed without merging**